source/val/construct.h

```diff
@@ -123,4 +123,15 @@ class Construct {
 
 }  /// namespace libspirv
 
+// Override std::hash for libspirv::ConstructType so that the enum can be used
+// as a key in an unordered_map.
+namespace std {
```

Comment:
Style guide says don't override std::hash.
https://google.github.io/styleguide/cppguide.html#std_hash

Make your own class name for the is implementation.